File: solutions/r/darts/1/darts.R
score <- function(x, y) {
  rsq = x*x + y*y
  score<-if(rsq<=1){
    10 
  } else if(rsq<=25){
    5
  } else if (rsq<=100){
    1
  } else {
    0
  }
}

File: solutions/free-pascal/darts/1/Darts.pas
unit Darts;

{$mode ObjFPC}{$H+}

interface

function score(const x : single; const y : single) : uint8;

implementation

uses SysUtils;

function score(const x : single; const y : single) : uint8;
var r :  single;
begin
   r:=sqrt(x*x + y*y);
   result:=0;
   if r<=1 then
      result:=10
   else if r<=5 then
      result:=5
   else if r<=10 then
      result:=1;
end;

end.
